### PR TITLE
Add RAPIDS_BASE_BRANCH in custom-job.yaml.

### PR DESCRIPTION
**Pull request by bdice**
Adds `RAPIDS_BASE_BRANCH` to the custom job definition. Helps with https://github.com/rapidsai/cugraph-ops/pull/426/

cc: @MatthiasKohl

### CODE REVIEW

**Comment by bdice** [comment]
Can you please test in your PR? Test with `uses: rapidsai/shared-action-workflows/.github/workflows/checks.yaml@custom-job-base-branch`. Then we can merge this and change your tag back to `branch-23.04`.

File: .github/workflows/custom-job.yaml [hunk 1]
```diff
@@ -66,11 +66,15 @@ jobs:
...
         run: |
+          echo "RAPIDS_BASE_BRANCH=${{ fromJSON(steps.get-pr-info.outputs.pr-info).base.ref }}" >> "${GITHUB_ENV}"
+          echo "RAPIDS_NIGHTLY_DATE=${{ inputs.date }}" >> "${GITHUB_ENV}"
+          echo "RAPIDS_REF_NAME=${{ inputs.branch || github.ref_name }}" >> "${GITHUB_ENV}"
           echo "RAPIDS_REPOSITORY=${{ inputs.repo || github.repository }}" >> "${GITHUB_ENV}"
           echo "RAPIDS_SHA=$(git rev-parse HEAD)" >> "${GITHUB_ENV}"
-          echo "RAPIDS_REF_NAME=${{ inputs.branch || github.ref_name }}" >> "${GITHUB_ENV}"
-          echo "RAPIDS_NIGHTLY_DATE=${{ inputs.date }}" >> "${GITHUB_ENV}"
       - name: Run script
         run: ${{ inputs.run_script }}
```
[the remark this answers is not in the document]
Comment (MatthiasKohl) [reply .github/workflows/custom-job.yaml:80]: ahh I didn't understand the environment variables above this, so yeah this isn't needed

**Review by ajschmidt8 — APPROVED** [review]

File: .github/workflows/custom-job.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (bdice) [reply .github/workflows/custom-job.yaml:80]: I took a chance and hoped this would work the same way. Seems like it’s fine. Thanks for testing.

**Comment by bdice** [comment]
Looks like things are working downstream with @ajschmidt8 's changes. Thanks so much AJ! https://github.com/rapidsai/cugraph-ops/actions/runs/4450499685/jobs/7816055390

**Pull request merged**